### ISSUE TITLE
[Bug] Big difference between the inference results (segmentation) of OpenVINO and TensorFlow.

### ISSUE THREAD

**Issue by superhez** [openvinotoolkit/openvino#5736]
##### System information (version)
- OpenVINO => 2021.3.394
- Operating System / Platform => Windows 10 Pro 64 Bit (20H2)
- Problem classification: Inference
- Framework: TensorFlow 2.2.0
- Model name: Customized decoder-encoder NN for segmentation (I think all layers are supported according to the documents.)

##### Detailed description
I trained and saved a non-frozen model in TensorFlow v2.2 with nvidia GPU.  Then I brought the model to a CPU environment with the same TensorFlow version and froze the model to .pb file (with a "variables" folder and a empty "assets" folder), and converted it to IR model using below command.

`python mo_tf.py  --saved_model_dir C:\EvaluationVer01\model\pb\  --input_shape [1,320,320,1]`

No error happened during all the processes. However, the inference results of OpenVINO are greatly different from that of TensorFlow (what it should be). Our model accepts a single channel 16-bit input image. We transfer it to float32 by normalization before feed to model.  The whole mode is in the precision float32. We have checked all layers in the DL workbench model visualization.

SORRY for the inconvenience to post the codes here. Instead, the layers used in our model are listed below.
tf.keras.layers.Conv2D
tf.keras.layers.BatchNormalization
tf.keras.layers.Activation
tf.keras.layers.Add()
tf.keras.layers.Conv2DTranspose
tf.nn.avg_pool2d
tf.concat

The first image is produced by TensorFlow (almost black but correct) and the second one is by OpenVINO(full of noise)
**TensorFlow**
![sample_tf](https://user-images.githubusercontent.com/33442089/119090278-66329480-ba46-11eb-8b0c-31ed095bb3fd.jpg)
**OpenVINO**
![sample_vino](https://user-images.githubusercontent.com/33442089/119090290-6894ee80-ba46-11eb-9b82-f2aece04ed1d.jpg)

##### Issue submission checklist

 - [ ✓] I report the issue, it's not a question
   
 - [ ✓] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ×] There is reproducer code and related data files: images, videos, models, etc.

**Comment by ilya-lavrenov** [comment]
@superhez in which layout are you providing a data? MO for TF converts NHWC layout to NCHW, please keep this in mind.

**Comment by superhez** [comment]
> @superhez in which layout are you providing a data? MO for TF converts NHWC layout to NCHW, please keep this in mind.

Thank you for advice. I have changed the layout of input to NCHW when use OpenVINO's inference engine. (otherwise error occurs)

**Comment by ilyachur** [comment]
@superhez Was the issue solved?

**Comment by superhez** [comment]
> @superhez Was the issue solved?
No... now I am about to confirm the outputs of layers to find from which layers wrong results arises.

**Comment by zulkifli-halim** [comment]
Hi, do still need help on this issue?

**Comment by superhez** [comment]
> Hi, do still need help on this issue?
Yes... So far no solution has been found yet. Since it is difficult to analyze the output of each layer, we have stopped it. Meanwhile, as we realized several layer functions may be not fully supported,  we changed them as below but no improvement is observed.

・tf.keras.layers.Activation　→　tf.keras.layers.ReLU　
・tf.nn.avg_pool2d　→　tf.keras.layers.AveragePooling2D
・tf.concat　→　tf.keras.layers.Concatenate

Actually, we succeeded to convert a similar (also a pix2pix NN but not totally same) model and performed the inference in previous experiments. But at that time, the versions we used were 2020.1(OpenVINO) and 1.14(TensorFlow). So I am wondering whether it could be solved if I return the frameworks to the old versions above. I would be grateful it if you could provide some more hints and confirmation items.

**Comment by jgespino** [comment]
Hi @superhez could you please test on the latest OpenVINO 2021.4 release and let me know if observe the same behavior?

**Comment by superhez** [comment]
> Hi @superhez could you please test on the latest OpenVINO 2021.4 release and let me know if observe the same behavior?

Thanks for your information. I tried OpenVINO 2021.4 today but it gave same results as 2021.3. Now I am returning TensorFlow to 1.14 and OpenVINO to 2020.1 to check if the combination works.

**Comment by jgespino** [comment]
@superhez Ok, please let me know if that solves your issue. If you still need assistance, please provide your model, code, input image and steps to reproduce. 

Regards,
Jesus
